[quantization] Modernize graph mode IR API calls

jamesr66a opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #30140 [quantization] Make observed values actually flow through observers
* **#30130 [quantization] Modernize graph mode IR API calls**


1. Use insertGetAttr instead of rolling it ourselves
2. Use schema-driven insertion for inserting observer calls
3. Don't initialize Module if we're just going to immediately overwrite it

Differential Revision: [D18608004](https://our.internmc.facebook.com/intern/diff/D18608004)

Commit:
[quantization] Modernize graph mode IR API calls

[ghstack-poisoned]

Commit:
Update on "[quantization] Modernize graph mode IR API calls"


Differential Revision: [D18608004](https://our.internmc.facebook.com/intern/diff/D18608004)

[ghstack-poisoned]

ZolotukhinM reviewed (approved):
Nice, thanks!

Commit:
Update on "[quantization] Modernize graph mode IR API calls"



1. Use insertGetAttr instead of rolling it ourselves
2. Use schema-driven insertion for inserting observer calls
3. Don't initialize Module if we're just going to immediately overwrite it

Differential Revision: [D18608004](https://our.internmc.facebook.com/intern/diff/D18608004)

[ghstack-poisoned]

jerryzh168 commented on torch/csrc/jit/passes/quantization.cpp:
```diff
@@ -198,25 +198,26 @@ Node* InsertObserversHelper::insertObserverFor(
-  call->insertAfter(observer_instance);
-  return call;
+
+  {
```
btw why do you put this inside a block?

jamesr66a replied:
`WithInsertPoint` is an RAII guard. As a rule I add a separate block when that's used, to ensure it goes out of scope at the right time, even if someone in the future adds code after the `return` here

Commit:
Update on "[quantization] Modernize graph mode IR API calls"



1. Use insertGetAttr instead of rolling it ourselves
2. Use schema-driven insertion for inserting observer calls
3. Don't initialize Module if we're just going to immediately overwrite it

Differential Revision: [D18608004](https://our.internmc.facebook.com/intern/diff/D18608004)

[ghstack-poisoned]
